tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -1691,6 +1714,90 @@ class LocationSet {
   uint32_t nextLoc[kMaxIndex];              ///< Next available location
 };
 
+/// A class for managing stage input/output packed locations to avoid duplicate
```

Comment:
Can this class be moved to its own file? I know `DeclResultIdMapper` is already out of hand but still better not to make it worse.